source/features/show-asset-download-count.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import {v4, escapeKey} from '../libs/api';
+import {cloudDownload} from '../libs/icons';
+import {getOwnerAndRepo, getRepoPath} from '../libs/utils';
+import {wrap} from '../libs/dom-utils';
+
+interface Asset {
+	name: string;
+	downloadCount: number;
+}
+interface Tag {
+	[key: string]: Asset[];
+}
+async function getAssetsForTag(tags: string[]): Promise<Tag> {
+	const {ownerName, repoName} = getOwnerAndRepo();
+	const data = await v4(
+		'{' +
+			tags.map(tag =>
+				escapeKey(tag) + `: repository(owner: "${ownerName}", name: "${repoName}") { release(tagName:"${tag}") { releaseAssets(first: 100) { edges { node { name downloadCount } } } } }`
+			) +
+		'}'
+	);
+	const assets: Tag = {};
+	Object.entries(data).forEach(([tag, repo]) => {
+		assets[tag] = repo.release.releaseAssets.edges.map((edge: any) => edge.node);
+	});
+	return assets;
+}
+
+async function init(): Promise<void | false> {
+	if (getRepoPath() === null || !(getRepoPath() as string).startsWith('releases')) {
+		return;
+	}
+
+	const tags = select.all('svg.octicon-tag ~ span').map(tag => tag.innerText);
+	const tagAssets = await getAssetsForTag(tags);
+	for (const release of select.all('.release-entry:not(.release-timeline-tags)')) {
```

Comment:
Instead of selecting again, it may be best to reuse the existing `tags` to ensure they point to the same stuff.

Reply:
I'm not quite sure I'm following

Reply:
On line 37 and 39 you’re selecting tags and releases, once to get the tag name and the second time to fill the asset list. 

You can do two things:

- select the releases at the beginning, and not the tags
- reuse the same array to reach the asset list of each release

Also it may be worth to select only the releases that have extra assets and then avoid the query altogether if there aren’t any.

Reply:
Given the dom layout, I still don't think you can select just the tags once:

```
  Release
     |
 |--------|
Tag     Assets
```

If you select the tags only once, then you lose your only connection to the assets (releases).

```
tags = select.all()
downloadCounts = await api(tags)
for tag of tags
  Add downloadCounts[tag] to tag // What assets belong to this tag? Yeah i might know the name but releases have the same assets in each (Electrons's SHASUMS256.txt)
endfor
```

Reply:
No, `tag` will still include the name (via `tag.textContent`) and you can use that to find the tag's assets in `downloadCounts`

Reply:
Also the tag selector needs to be updated to ensure it only selects tags that actually have assets, so we don't trigger a request even when it's not necessary (like on most Releases pages)

Reply:
Friendly ping on this PR 🥐

Reply:
@bfred-it Sorry, if I'm not getting this correctly but I still can't figure out how you can associate an asset with its correct tag without selecting the tag again from a common parent. Maybe I'm over thinking this.

Reply:
Alright, if you can address the other two points (indentation and pretty-bytes) I'll take a stab at this later

Reply:
The numbers are actually just raw numbers (download count) and not bytes, GitHub already displays the file size and pretty prints them.